### ISSUE TITLE
[ActiveRecord] link() should put newly created element in main model too

### ISSUE THREAD

**Issue by mikk150** [yiisoft/yii2#9615]
link() should put newly created element in main model too

**Comment by samdark** [comment]
Could you elaborate?

**Comment by mikk150** [comment]
Say i create new model, and link it with some other model, then new model should be pushed to the other models relatedRecords array.

**Comment by dynasource** [comment]
on success, that would indeed be a logical procedure.

**Comment by cebe** [comment]
you mean the relation defined via `inverseOf`?

**Comment by lynicidn** [comment]
i think talk about may to many

**Comment by mikk150** [comment]
consider code:
```php
$model=Model::find()->with('relation')->where(['id' => 10])->one();

$linkedElement=new LinkedModel();
$linkedElement->link('related', $model);
```
model related should have(according to has one or has many) either $model->relation===$linkedElement or array_search($linkedElement,$model->relation) == true

**Comment by cebe** [comment]
isn't this already the case? https://github.com/yiisoft/yii2/blob/99fc362e108aba3e8bd3e913de29d0e0ae2cdf80/framework/db/BaseActiveRecord.php#L1233-L1243

**Comment by mikk150** [comment]
this populates $linkedElement's _related array, but not $model _related array

**Comment by lynicidn** [comment]
hm, you want break related logic, look:
$post->link('user', $user);
$user->posts // return one post, relation populated and query not execute

**Comment by cebe** [comment]
I still don't really get it, can you add the Models including relation definition for your example code above?

**Comment by dynasource** [comment]
what he means is vice versa. Currently only one direction is implemented witin $this.

This will cause significant extra complexity. It requires:
* external setting of _related 
* a distinction between HAS_ONE and HAS_MANY 
* some mechanism for reverse relations such as inverseOf

**Comment by mikk150** [comment]
I used to (ab)use addRelatedRecord in Yii1 for same thing

**Comment by mikk150** [comment]
maybe (ab)use populateRelation?

**Comment by samdark** [comment]
Won't be implemented in 2.0.
